modules/navier_stokes/src/auxkernels/INSQCriterionAux.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSQCriterionAux.h"
+
+registerMooseObject("NavierStokesApp", INSQCriterionAux);
+
+InputParameters
+INSQCriterionAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+
+  params.addClassDescription("This class computes the Q criterion, a scalar which"
+                             "aids in vortex identification in turbulent flows");
+  params.addCoupledVar("velocity", "The velocity component");
+  return params;
+}
+
+INSQCriterionAux::INSQCriterionAux(const InputParameters & parameters)
+  : AuxKernel(parameters), _grad_velocity(coupledVectorGradient("velocity"))
+{
+}
+
+Real
+INSQCriterionAux::computeValue()
+{
+  const Real symm_part = 2.0 * std::pow(_grad_velocity[_qp](0, 0), 2) +
+                         2.0 * std::pow(_grad_velocity[_qp](1, 1), 2) +
+                         2.0 * std::pow(_grad_velocity[_qp](2, 2), 2) +
+                         std::pow(_grad_velocity[_qp](0, 2) + _grad_velocity[_qp](2, 0), 2) +
+                         std::pow(_grad_velocity[_qp](0, 1) + _grad_velocity[_qp](1, 0), 2) +
+                         std::pow(_grad_velocity[_qp](1, 2) + _grad_velocity[_qp](2, 1), 2);
+  const Real antisymm_part = std::pow(_grad_velocity[_qp](0, 2) - _grad_velocity[_qp](2, 0), 2) +
+                             std::pow(_grad_velocity[_qp](0, 1) - _grad_velocity[_qp](1, 0), 2) +
+                             std::pow(_grad_velocity[_qp](1, 2) - _grad_velocity[_qp](2, 1), 2);
```

Comment:
For consistency with the kernel (and for a slight speed boost), do you want to use `Utility::pow` here? I don't really care

Reply:
yeah, may as well! I checked the timing of the jacobian assembly before and after switching to the libmesh utility function, and the speedup was negligible. It's truly that sparsity operation that's killer here.